resurfemg/helper_functions.py

```diff
@@ -1876,38 +1844,29 @@ def sampen_optimized(
     return saen
 
 
-def calc_closed_sampent(emb_dim, t_vecs):
-    counts = []
-    for m in (emb_dim, emb_dim + 1):
-        counts.append(0)
-        # get the matrix that we need for the current m
-        t_vecs_m = t_vecs[:n - m + 1, :m]
-        # successively calculate distances between each pair of templ vectrs
-        for i in range(len(t_vecs_m) - 1):
-            dsts = dist(t_vecs_m[i + 1:], t_vecs_m[i])
-            # count how many distances are smaller than the tolerance
-            counts[-1] += np.sum(dsts <= tolerance)
-    return counts
+def calc_closed_sampent(t_vecs, n, tolerance):
+    # TODO(someone?): Analogous to calc_open_sampent
+    return np.nan, np.nan
 
 
-def calc_open_sampent(emb_dim, t_vecs):
-    # TODO
-    counts = []
-    for m in [emb_dim, emb_dim + 1]:
-        counts.append(0)
-        # get the matrix that we need for the current m
-        t_vecs_m = t_vecs[:n - m + 1, :m]
-        # successively calculate distances between each pair of templ vectrs
-        for i in range(len(t_vecs_m) - 1):
-            dsts = dist(t_vecs_m[i + 1:], t_vecs_m[i])
-            # count how many distances are smaller than the tolerance
-            counts[-1] += np.sum(dsts < tolerance)
-    return counts
+def calc_open_sampent(t_vecs, n, tolerance):
+    triplets = t_vecs[:n - 2, :3]
+
+    raw_dsts = tuple(
+        triplets[i + 1:] - triplets[i]
+        for i in range(len(triplets) - 1)
+    )
+    dsts = np.concatenate(raw_dsts)
+    dsts_abs = np.abs(dsts)
+    dsts_gt = dsts_abs < tolerance
+    dsts_max_a = np.logical_and(dsts_gt[:, 0], dsts_gt[:, 1])
+    dsts_max = np.logical_and(dsts_max_a, dsts_gt[:, 2])
+    return np.sum(dsts_max_a), np.sum(dsts_max)
 
 
 def entropy_maker(
         array,
-        method='nolds',
+        method='nolds_optimized',
```

Comment:
the naming is a bit confusing. I would use 'sample_entropy', but this is a minor.